core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -34,6 +34,9 @@ class SearchService(
     with FileChangeListener
     with SLF4JLogging {
 
+  private[indexer] def isUserFile(file: File): Boolean =
+    config.allRoots exists (r => file descendsFrom r)
```

Comment:
I don't understand how this works. `allRoots` are source files, but isn't `file: File` a binary class file or a jar? It would be better to take a solid type here instead of `File`.